### PR TITLE
Added information about versioning

### PR DESCRIPTION
**Pull request by juergenchrist**
README.md now includes some guidelines on how to update the version number.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,4 +18,26 @@ If no options or command is specified, `mx` prints information on the available
...
+`mx` uses a `major`.`minor`.`patch` versioning scheme.  To figure out if you
+version is sufficient for a given `mx` suite, first compare the major version
+number of your `mx` version against the major number of the required version
+specified in the suite.  If these versions are not equal, you cannot expect
+`mx` to be compatible with this suite.  The minor number has to be greater or
+equal to the specified minor version number.  Compatibility is ensured
+regardless of the patch level.  However, if your patch level is lower than the
+required patch level you might trigger bugs in `mx`.
+
+From a developer point of view this versioning scheme enforces the following
```
Comment (dougxc) [inline README.md]: from a mx developer point of view

File: README.md [hunk 2]
```diff
@@ -18,4 +18,26 @@ If no options or command is specified, `mx` prints information on the available
 
 For an example of `mx` usage, you can read the [Instructions][1] for the Graal project. 
 
+### mx versioning ###
+
+`mx` uses a `major`.`minor`.`patch` versioning scheme.  To figure out if you
```
Comment (dougxc) [inline README.md]: To figure out if **the** version

**Comment by dougxc** [comment]
lgtm modulo fixes for above comments

**Comment by rschatz** [comment]
Does that mean that *any* change to mx bumps the patch version number? I always was under the impression that pure bugfixes that have no impact on the interface don't need a version bump...

File: README.md [hunk 3]
```diff
@@ -18,4 +18,26 @@ If no options or command is specified, `mx` prints information on the available
...
+`mx` uses a `major`.`minor`.`patch` versioning scheme.  To figure out if the
+version is sufficient for a given `mx` suite, first compare the major version
+number of your `mx` version against the major number of the required version
+specified in the suite.  If these versions are not equal, you cannot expect
+`mx` to be compatible with this suite.  The minor number has to be greater or
+equal to the specified minor version number.  Compatibility is ensured
+regardless of the patch level.  However, if your patch level is lower than the
+required patch level you might trigger bugs in `mx`.
+
+From a `mx` developer point of view this versioning scheme enforces the following
```
Comment (woess) [inline README.md]: shouldn't it be **an** mx developer?

Comment (dougxc) [inline README.md]: I agree that "an mx" sounds better but I believe "a mx" is [correct](http://writingexplained.org/a-vs-an-difference)

**Comment by dougxc** [comment]
It seems hard to come up with a clear policy when to bump the patch level. I think the rough distinction of bug fix versus other minor fixes such as typos fixes, documentation update, private variable renaming variable may be useful. We should increase patch number for the former but not necessarily for the latter.

File: README.md (hunk 3 above)
Comment (rschatz) [inline README.md]: Actually, the link you posted says exactly the opposite:
> An is used before words, abbreviations, acronyms, or letters that begin with a **vowel sound**, regardless of their spelling

They even give an example that's probably as close as you can get before "mx" finally gets accepted into dictionaries ;)
>A Microsoft program.
>-but-
>An MS-DOS program.

**Pull request merged**